File: flutter17_mobile/lib/models/user_account.dart
import 'package:flutter17_mobile/models/customer.dart';
import 'package:flutter17_mobile/models/image.dart';
import 'package:flutter17_mobile/models/role.dart';
import 'package:flutter17_mobile/models/seller.dart';
import 'package:json_annotation/json_annotation.dart';

part 'user_account.g.dart';

@JsonSerializable()
class UserAccount {
  int? id;
  String? username;
  String? email;
  DateTime? registrationDate;
  Customer? customer;
  Seller? seller;
  Role? role;
  Image? image;
  bool? isDeactivated;

  UserAccount({
    this.id,
    this.username,
    this.email,
    this.registrationDate,
    this.customer,
    this.seller,
    this.role,
    this.image,
    this.isDeactivated,
  });

  factory UserAccount.fromJson(Map<String, dynamic> json) => _$UserAccountFromJson(json);

  Map<String, dynamic> toJson() => _$UserAccountToJson(this);
}

File: angular17_web/src/app/models/user-account/user.account.ts
import { Customer } from '../customer/customer';
import { Seller } from '../seller/seller';
import { Role } from '../role/role';
import { Image } from '../image/image';

export class UserAccount {
  id!: number;
  username!: string;
  email!: string;
  registrationDate!: Date;
  customer?: Customer;
  seller?: Seller;
  role!: Role;
  image!: Image;
  isDeactivated!: boolean;

  constructor(data?: any) {
    if (!data) return;

    this.id = data.id;
    this.username = data.username;
    this.email = data.email;
    this.registrationDate = data.registrationDate ? new Date(data.registrationDate) : undefined as any;
    this.customer = data.customer ? new Customer(data.customer) : undefined;
    this.seller = data.seller ? new Seller(data.seller) : undefined;
    this.role = data.role ? new Role(data.role) : undefined as any;
    this.image = data.image ? new Image(data.image) : undefined as any;
    this.isDeactivated = data.isDeactivated;
  }
}
